src/query/functions/temporal/base.go

```diff
@@ -279,11 +308,28 @@ func buildBlockBatch(
 }
 
 func singleProcess(
+	ctx context.Context,
 	seriesIter block.SeriesIter,
 	builder block.Builder,
 	m blockMeta,
 	p processor,
 ) error {
+	var (
+		start      = time.Now()
+		decodeTime time.Duration
+	)
+	_, sp, _ := xcontext.StartSampledTraceSpan(ctx, tracepoint.TemporalDecodeSingle)
```

Comment:
Should probably move this after the `decodeTime == 0` check in the defer function, similarly to how `parallelProcess` does it